lib/stdlib/src/gen_server.erl

```diff
@@ -2243,33 +2243,62 @@ init_it(Mod, Args) ->
 
 loop(ServerData, State, {continue, Continue} = Msg, Debug) ->
     Reply = try_handle_continue(ServerData, State, Continue),
+    From  = undefined,
     case Debug of
         [] ->
-            handle_common_reply(ServerData, State, Msg, undefined, Reply);
+            handle_common_reply(ServerData, State, Msg, From, Reply);
         _ ->
             Debug1 = sys:handle_debug(Debug, fun print_event/3, ServerData#server_data.name, Msg),
-            handle_common_reply(ServerData, State, Msg, undefined, Reply, Debug1)
+            handle_common_reply(ServerData, State, Msg, From, Reply, Debug1)
     end;
+%%
+loop(ServerData, State, LoopAction, Debug) ->
+    case LoopAction of
+        {timeout_zero, TimeoutMsg} ->
+            decode_msg(ServerData, State, TimeoutMsg, undefined, Debug, false);
+        {timeout, TRef, HibInf} ->
+            loop(ServerData, State, HibInf, TRef, Debug);
+        HibT ->
+            loop(ServerData, State, HibT, undefined, Debug)
+    end.
```

Comment:
This will do some argument shuffling since `Debug` goes from 4th to 5th argument. If the `LoopAction` and `Debug` order was flipped, this wouldn't be necessary.

Reply:
That could be worth changing.  It also hits `loop/5`, but I don't find any call that goes against that argument order.